2024-08-12 11:13:02,345 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 11:13:25,559 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=-1&timeout=20 HTTP/11" 200 23
2024-08-12 11:13:25,660 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getMe HTTP/11" 200 263
2024-08-12 11:13:25,670 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 11:13:46,378 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=1&timeout=20 HTTP/11" 200 23
2024-08-12 11:13:48,367 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=1&timeout=20 HTTP/11" 200 361
2024-08-12 11:13:48,375 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 11:13:48,991 - DEBUG - https://api.telegram.org:443 "POST /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/sendMessage?chat_id=245324256&text=Welcome+to+the+ECommerce+bot%21+Use+%2Fhelp+to+see+available+commands. HTTP/11" 200 380
2024-08-12 11:13:53,355 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222752&timeout=20 HTTP/11" 200 360
2024-08-12 11:13:53,360 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 11:13:53,968 - DEBUG - https://api.telegram.org:443 "POST /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/sendMessage?chat_id=245324256&text=Available+commands%3A%0A%2Fstart+-+Welcome+message+and+basic+instructions%0A%2Fhelp+-+Display+this+help+message%0A%2Flist_products+%3Ccategory%3E+-+List+products+by+category+%28e.g.%2C+%2Flist_products+shirts%29%0A%2Fproduct_detail+%3Cproduct_id%3E+-+Product+detail+by+ID+%28e.g.%2C+%2Flist_products+product_id%29%0A%2Fview_cart+-+View+your+shopping+cart%0A%2Fadd_to_cart+%3Cproduct_id%3E+%3Cquantity%3E+-+Add+a+product+to+your+cart%0A%2Fremove_from_cart+%3Cproduct_id%3E+%3Cquantity%3E+-+Add+a+product+to+your+cart%0A%2Fprofile+-+View+or+update+your+profile+information%0A HTTP/11" 200 1247
2024-08-12 11:13:59,431 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222753&timeout=20 HTTP/11" 200 370
2024-08-12 11:14:03,650 - ERROR - Infinity polling exception: 2003 (HY000): Can't connect to MySQL server on 'localhost:3306' (10061)
2024-08-12 11:14:03,650 - ERROR - Exception traceback:
Traceback (most recent call last):
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\mysql\connector\connection_cext.py", line 334, in _open_connection
    self._cmysql.connect(**cnx_kwargs)
_mysql_connector.MySQLInterfaceError: Can't connect to MySQL server on 'localhost:3306' (10061)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1090, in infinity_polling
    self.polling(non_stop=True, timeout=timeout, long_polling_timeout=long_polling_timeout,
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1178, in polling
    self.__threaded_polling(non_stop=non_stop, interval=interval, timeout=timeout, long_polling_timeout=long_polling_timeout,
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1253, in __threaded_polling
    raise e
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1215, in __threaded_polling
    self.worker_pool.raise_exceptions()
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\util.py", line 150, in raise_exceptions
    raise self.exception_info
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\util.py", line 93, in run
    task(*args, **kwargs)
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 8623, in _run_middlewares_and_handler
    result = handler['function'](message)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\Personal Computer\Desktop\Ecommerce bot\ecommerce\handlers\commands.py", line 29, in list_products_command
    products = get_all_products()
               ^^^^^^^^^^^^^^^^^^
  File "c:\Users\Personal Computer\Desktop\Ecommerce bot\ecommerce\utils\database_utils.py", line 18, in get_all_products
    conn = get_db_connection()
           ^^^^^^^^^^^^^^^^^^^
  File "c:\Users\Personal Computer\Desktop\Ecommerce bot\ecommerce\utils\database_utils.py", line 6, in get_db_connection
    conn = mysql.connector.connect(**DB_CONFIG)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\mysql\connector\pooling.py", line 322, in connect
    return CMySQLConnection(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\mysql\connector\connection_cext.py", line 151, in __init__
    self.connect(**kwargs)
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\mysql\connector\abstracts.py", line 1399, in connect
    self._open_connection()
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\mysql\connector\connection_cext.py", line 339, in _open_connection
    raise get_mysql_exception(
mysql.connector.errors.DatabaseError: 2003 (HY000): Can't connect to MySQL server on 'localhost:3306' (10061)

2024-08-12 11:14:06,668 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 11:14:07,230 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222754&timeout=20 HTTP/11" 409 143
2024-08-12 11:14:27,230 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222754&timeout=20 HTTP/11" 200 23
2024-08-12 11:14:47,343 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222754&timeout=20 HTTP/11" 200 23
2024-08-12 11:15:00,425 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222754&timeout=20 HTTP/11" 200 370
2024-08-12 11:15:00,491 - ERROR - Infinity polling exception: 'price'
2024-08-12 11:15:00,492 - ERROR - Exception traceback:
Traceback (most recent call last):
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1090, in infinity_polling
    self.polling(non_stop=True, timeout=timeout, long_polling_timeout=long_polling_timeout,
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1178, in polling
    self.__threaded_polling(non_stop=non_stop, interval=interval, timeout=timeout, long_polling_timeout=long_polling_timeout,
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1253, in __threaded_polling
    raise e
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1215, in __threaded_polling
    self.worker_pool.raise_exceptions()
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\util.py", line 150, in raise_exceptions
    raise self.exception_info
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\util.py", line 93, in run
    task(*args, **kwargs)
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 8623, in _run_middlewares_and_handler
    result = handler['function'](message)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\Personal Computer\Desktop\Ecommerce bot\ecommerce\handlers\commands.py", line 33, in list_products_command
    response += f"ID: {product['product_id']}\nName: {product['name']}\nPrice: ${product['price']}\nInventory: {product['inventory']}\n\n"
                                                                                 ~~~~~~~^^^^^^^^^
KeyError: 'price'

2024-08-12 11:15:03,494 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 11:15:04,204 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222755&timeout=20 HTTP/11" 409 143
2024-08-12 11:15:24,206 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222755&timeout=20 HTTP/11" 200 23
2024-08-12 11:15:44,327 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222755&timeout=20 HTTP/11" 200 23
2024-08-12 11:16:04,454 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222755&timeout=20 HTTP/11" 200 23
2024-08-12 11:16:24,585 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222755&timeout=20 HTTP/11" 200 23
2024-08-12 11:16:44,713 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222755&timeout=20 HTTP/11" 200 23
2024-08-12 11:17:04,818 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222755&timeout=20 HTTP/11" 200 23
2024-08-12 11:17:24,966 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222755&timeout=20 HTTP/11" 200 23
2024-08-12 11:17:45,097 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222755&timeout=20 HTTP/11" 200 23
2024-08-12 11:18:05,194 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222755&timeout=20 HTTP/11" 200 23
2024-08-12 11:18:25,319 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222755&timeout=20 HTTP/11" 200 23
2024-08-12 11:18:45,466 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222755&timeout=20 HTTP/11" 200 23
2024-08-12 11:18:49,708 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222755&timeout=20 HTTP/11" 200 370
2024-08-12 11:18:49,735 - ERROR - Infinity polling exception: 1146 (42S02): Table 'ecommercebotdb.product' doesn't exist
2024-08-12 11:18:49,735 - ERROR - Exception traceback:
Traceback (most recent call last):
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\mysql\connector\connection_cext.py", line 705, in cmd_query
    self._cmysql.query(
_mysql_connector.MySQLInterfaceError: Table 'ecommercebotdb.product' doesn't exist

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1090, in infinity_polling
    self.polling(non_stop=True, timeout=timeout, long_polling_timeout=long_polling_timeout,
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1178, in polling
    self.__threaded_polling(non_stop=non_stop, interval=interval, timeout=timeout, long_polling_timeout=long_polling_timeout,
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1253, in __threaded_polling
    raise e
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1215, in __threaded_polling
    self.worker_pool.raise_exceptions()
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\util.py", line 150, in raise_exceptions
    raise self.exception_info
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\util.py", line 93, in run
    task(*args, **kwargs)
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 8623, in _run_middlewares_and_handler
    result = handler['function'](message)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\Personal Computer\Desktop\Ecommerce bot\ecommerce\handlers\commands.py", line 29, in list_products_command
    products = get_all_products()
               ^^^^^^^^^^^^^^^^^^
  File "c:\Users\Personal Computer\Desktop\Ecommerce bot\ecommerce\utils\database_utils.py", line 23, in get_all_products
    cursor.execute('SELECT product_id,name FROM product')
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\mysql\connector\cursor_cext.py", line 357, in execute
    result = self._connection.cmd_query(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\mysql\connector\opentelemetry\context_propagation.py", line 97, in wrapper
    return method(cnx, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\mysql\connector\connection_cext.py", line 713, in cmd_query
    raise get_mysql_exception(
mysql.connector.errors.ProgrammingError: 1146 (42S02): Table 'ecommercebotdb.product' doesn't exist

2024-08-12 11:18:52,748 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 11:18:53,343 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222756&timeout=20 HTTP/11" 409 143
2024-08-12 11:19:13,344 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222756&timeout=20 HTTP/11" 200 23
2024-08-12 11:19:14,830 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222756&timeout=20 HTTP/11" 200 360
2024-08-12 11:19:14,830 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 11:19:15,619 - DEBUG - https://api.telegram.org:443 "POST /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/sendMessage?chat_id=245324256&text=Available+commands%3A%0A%2Fstart+-+Welcome+message+and+basic+instructions%0A%2Fhelp+-+Display+this+help+message%0A%2Flist_products+%3Ccategory%3E+-+List+products+by+category+%28e.g.%2C+%2Flist_products+shirts%29%0A%2Fproduct_detail+%3Cproduct_id%3E+-+Product+detail+by+ID+%28e.g.%2C+%2Flist_products+product_id%29%0A%2Fview_cart+-+View+your+shopping+cart%0A%2Fadd_to_cart+%3Cproduct_id%3E+%3Cquantity%3E+-+Add+a+product+to+your+cart%0A%2Fremove_from_cart+%3Cproduct_id%3E+%3Cquantity%3E+-+Add+a+product+to+your+cart%0A%2Fprofile+-+View+or+update+your+profile+information%0A HTTP/11" 200 1247
2024-08-12 11:19:17,741 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222757&timeout=20 HTTP/11" 200 371
2024-08-12 11:19:17,761 - ERROR - Infinity polling exception: 1146 (42S02): Table 'ecommercebotdb.product' doesn't exist
2024-08-12 11:19:17,761 - ERROR - Exception traceback:
Traceback (most recent call last):
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\mysql\connector\connection_cext.py", line 705, in cmd_query
    self._cmysql.query(
_mysql_connector.MySQLInterfaceError: Table 'ecommercebotdb.product' doesn't exist

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1090, in infinity_polling
    self.polling(non_stop=True, timeout=timeout, long_polling_timeout=long_polling_timeout,
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1178, in polling
    self.__threaded_polling(non_stop=non_stop, interval=interval, timeout=timeout, long_polling_timeout=long_polling_timeout,
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1253, in __threaded_polling
    raise e
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1215, in __threaded_polling
    self.worker_pool.raise_exceptions()
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\util.py", line 150, in raise_exceptions
    raise self.exception_info
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\util.py", line 93, in run
    task(*args, **kwargs)
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 8623, in _run_middlewares_and_handler
    result = handler['function'](message)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\Personal Computer\Desktop\Ecommerce bot\ecommerce\handlers\commands.py", line 29, in list_products_command
    products = get_all_products()
               ^^^^^^^^^^^^^^^^^^
  File "c:\Users\Personal Computer\Desktop\Ecommerce bot\ecommerce\utils\database_utils.py", line 23, in get_all_products
    cursor.execute('SELECT product_id,name FROM product')
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\mysql\connector\cursor_cext.py", line 357, in execute
    result = self._connection.cmd_query(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\mysql\connector\opentelemetry\context_propagation.py", line 97, in wrapper
    return method(cnx, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\mysql\connector\connection_cext.py", line 713, in cmd_query
    raise get_mysql_exception(
mysql.connector.errors.ProgrammingError: 1146 (42S02): Table 'ecommercebotdb.product' doesn't exist

2024-08-12 11:19:20,777 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 11:19:21,345 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222758&timeout=20 HTTP/11" 409 143
2024-08-12 11:19:41,345 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222758&timeout=20 HTTP/11" 200 23
2024-08-12 11:20:01,453 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222758&timeout=20 HTTP/11" 200 23
2024-08-12 11:20:27,041 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 11:20:29,939 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=-1&timeout=20 HTTP/11" 200 371
2024-08-12 11:20:30,046 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getMe HTTP/11" 200 263
2024-08-12 11:20:30,050 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 11:20:30,632 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=1&timeout=20 HTTP/11" 200 371
2024-08-12 11:20:30,935 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 11:20:31,460 - DEBUG - https://api.telegram.org:443 "POST /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/sendMessage?chat_id=245324256&text=No+products+available. HTTP/11" 200 278
2024-08-12 11:20:50,746 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222759&timeout=20 HTTP/11" 200 23
2024-08-12 11:21:10,861 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222759&timeout=20 HTTP/11" 200 23
2024-08-12 11:21:30,985 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222759&timeout=20 HTTP/11" 200 23
2024-08-12 11:21:51,099 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222759&timeout=20 HTTP/11" 200 23
2024-08-12 11:22:11,212 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222759&timeout=20 HTTP/11" 200 23
2024-08-12 11:22:31,326 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222759&timeout=20 HTTP/11" 200 23
2024-08-12 11:22:44,658 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222759&timeout=20 HTTP/11" 200 371
2024-08-12 11:22:44,674 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 11:22:45,438 - DEBUG - https://api.telegram.org:443 "POST /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/sendMessage?chat_id=245324256&text=No+products+available. HTTP/11" 200 278
2024-08-12 11:23:04,789 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222760&timeout=20 HTTP/11" 200 23
2024-08-12 11:23:24,905 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222760&timeout=20 HTTP/11" 200 23
2024-08-12 11:23:45,657 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 11:23:54,645 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=-1&timeout=20 HTTP/11" 200 378
2024-08-12 11:23:54,774 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getMe HTTP/11" 200 263
2024-08-12 11:23:54,776 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 11:23:55,890 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=1&timeout=20 HTTP/11" 200 378
2024-08-12 11:23:55,901 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 11:23:56,558 - DEBUG - https://api.telegram.org:443 "POST /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/sendMessage?chat_id=245324256&text=No+products+available. HTTP/11" 200 278
2024-08-12 11:24:15,997 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222761&timeout=20 HTTP/11" 200 23
2024-08-12 11:24:36,100 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222761&timeout=20 HTTP/11" 200 23
2024-08-12 11:24:56,203 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222761&timeout=20 HTTP/11" 200 23
2024-08-12 11:25:16,299 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222761&timeout=20 HTTP/11" 200 23
2024-08-12 11:25:36,402 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222761&timeout=20 HTTP/11" 200 23
2024-08-12 11:25:56,522 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222761&timeout=20 HTTP/11" 200 23
2024-08-12 11:26:16,624 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222761&timeout=20 HTTP/11" 200 23
2024-08-12 11:30:14,720 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 11:30:27,817 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=-1&timeout=20 HTTP/11" 200 371
2024-08-12 11:30:27,922 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getMe HTTP/11" 200 263
2024-08-12 11:30:27,925 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 11:30:28,468 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=1&timeout=20 HTTP/11" 200 371
2024-08-12 11:30:28,475 - ERROR - Infinity polling exception: list index out of range
2024-08-12 11:30:28,475 - ERROR - Exception traceback:
Traceback (most recent call last):
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1090, in infinity_polling
    self.polling(non_stop=True, timeout=timeout, long_polling_timeout=long_polling_timeout,
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1178, in polling
    self.__threaded_polling(non_stop=non_stop, interval=interval, timeout=timeout, long_polling_timeout=long_polling_timeout,
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1253, in __threaded_polling
    raise e
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1215, in __threaded_polling
    self.worker_pool.raise_exceptions()
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\util.py", line 150, in raise_exceptions
    raise self.exception_info
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\util.py", line 93, in run
    task(*args, **kwargs)
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 8623, in _run_middlewares_and_handler
    result = handler['function'](message)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\Personal Computer\Desktop\Ecommerce bot\ecommerce\handlers\commands.py", line 31, in list_products_command
    print(parts[1])
          ~~~~~^^^
IndexError: list index out of range

2024-08-12 11:30:31,493 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 11:30:32,045 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222762&timeout=20 HTTP/11" 409 143
2024-08-12 11:31:02,444 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 11:31:03,109 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=-1&timeout=20 HTTP/11" 200 371
2024-08-12 11:31:03,219 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getMe HTTP/11" 200 263
2024-08-12 11:31:03,219 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 11:31:03,760 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=1&timeout=20 HTTP/11" 200 371
2024-08-12 11:31:03,767 - ERROR - Infinity polling exception: 'price'
2024-08-12 11:31:03,772 - ERROR - Exception traceback:
Traceback (most recent call last):
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1090, in infinity_polling
    self.polling(non_stop=True, timeout=timeout, long_polling_timeout=long_polling_timeout,
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1178, in polling
    self.__threaded_polling(non_stop=non_stop, interval=interval, timeout=timeout, long_polling_timeout=long_polling_timeout,
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1253, in __threaded_polling
    raise e
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1215, in __threaded_polling
    self.worker_pool.raise_exceptions()
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\util.py", line 150, in raise_exceptions
    raise self.exception_info
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\util.py", line 93, in run
    task(*args, **kwargs)
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 8623, in _run_middlewares_and_handler
    result = handler['function'](message)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\Personal Computer\Desktop\Ecommerce bot\ecommerce\handlers\commands.py", line 35, in list_products_command
    response += f"ID: {product['product_id']}\nName: {product['name']}\nPrice: ${product['price']}\nInventory: {product['inventory']}\n\n"
                                                                                 ~~~~~~~^^^^^^^^^
KeyError: 'price'

2024-08-12 11:31:06,179 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222763&timeout=20 HTTP/11" 200 371
2024-08-12 11:31:06,780 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 11:31:06,776 - ERROR - Infinity polling exception: 'price'
2024-08-12 11:31:06,783 - ERROR - Exception traceback:
Traceback (most recent call last):
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1090, in infinity_polling
    self.polling(non_stop=True, timeout=timeout, long_polling_timeout=long_polling_timeout,
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1178, in polling
    self.__threaded_polling(non_stop=non_stop, interval=interval, timeout=timeout, long_polling_timeout=long_polling_timeout,
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1253, in __threaded_polling
    raise e
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1215, in __threaded_polling
    self.worker_pool.raise_exceptions()
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\util.py", line 150, in raise_exceptions
    raise self.exception_info
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\util.py", line 93, in run
    task(*args, **kwargs)
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 8623, in _run_middlewares_and_handler
    result = handler['function'](message)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\Personal Computer\Desktop\Ecommerce bot\ecommerce\handlers\commands.py", line 35, in list_products_command
    response += f"ID: {product['product_id']}\nName: {product['name']}\nPrice: ${product['price']}\nInventory: {product['inventory']}\n\n"
                                                                                 ~~~~~~~^^^^^^^^^
KeyError: 'price'

2024-08-12 11:31:09,786 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 11:31:10,355 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222764&timeout=20 HTTP/11" 409 143
2024-08-12 11:31:30,358 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222764&timeout=20 HTTP/11" 200 23
2024-08-12 11:31:49,827 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 11:31:52,680 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=-1&timeout=20 HTTP/11" 200 371
2024-08-12 11:31:52,822 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getMe HTTP/11" 200 263
2024-08-12 11:31:52,822 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 11:31:53,371 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=1&timeout=20 HTTP/11" 200 371
2024-08-12 11:31:53,389 - ERROR - Infinity polling exception: 'price'
2024-08-12 11:31:53,389 - ERROR - Exception traceback:
Traceback (most recent call last):
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1090, in infinity_polling
    self.polling(non_stop=True, timeout=timeout, long_polling_timeout=long_polling_timeout,
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1178, in polling
    self.__threaded_polling(non_stop=non_stop, interval=interval, timeout=timeout, long_polling_timeout=long_polling_timeout,
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1253, in __threaded_polling
    raise e
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1215, in __threaded_polling
    self.worker_pool.raise_exceptions()
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\util.py", line 150, in raise_exceptions
    raise self.exception_info
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\util.py", line 93, in run
    task(*args, **kwargs)
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 8623, in _run_middlewares_and_handler
    result = handler['function'](message)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\Personal Computer\Desktop\Ecommerce bot\ecommerce\handlers\commands.py", line 35, in list_products_command
    response += f"ID: {product['product_id']}\nName: {product['name']}\nPrice: ${product['price']}\nInventory: {product['inventory']}\n\n"
                                                                                 ~~~~~~~^^^^^^^^^
KeyError: 'price'

2024-08-12 11:31:56,398 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 11:31:56,948 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222765&timeout=20 HTTP/11" 409 143
2024-08-12 11:33:34,075 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 11:33:54,887 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=-1&timeout=20 HTTP/11" 200 23
2024-08-12 11:33:55,037 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getMe HTTP/11" 200 263
2024-08-12 11:33:55,037 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 11:34:48,211 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 11:34:48,876 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=-1&timeout=20 HTTP/11" 200 371
2024-08-12 11:34:48,988 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getMe HTTP/11" 200 263
2024-08-12 11:34:48,988 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 11:34:49,527 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=1&timeout=20 HTTP/11" 200 371
2024-08-12 11:34:49,542 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 11:34:49,791 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222766&timeout=20 HTTP/11" 200 371
2024-08-12 11:34:49,798 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 11:34:50,104 - DEBUG - https://api.telegram.org:443 "POST /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/sendMessage?chat_id=245324256&text=Available+products%3A%0AID%3A+1%0AName%3A+Graphic+T-Shirt%0APrice%3A+%2419.99%0AInventory%3A+50%0A%0AID%3A+2%0AName%3A+Slim+Fit+Jeans%0APrice%3A+%2439.99%0AInventory%3A+30%0A%0AID%3A+3%0AName%3A+Running+Sneakers%0APrice%3A+%2459.99%0AInventory%3A+20%0A%0A HTTP/11" 200 459
2024-08-12 11:34:50,371 - DEBUG - https://api.telegram.org:443 "POST /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/sendMessage?chat_id=245324256&text=Available+products%3A%0AID%3A+1%0AName%3A+Graphic+T-Shirt%0APrice%3A+%2419.99%0AInventory%3A+50%0A%0AID%3A+2%0AName%3A+Slim+Fit+Jeans%0APrice%3A+%2439.99%0AInventory%3A+30%0A%0AID%3A+3%0AName%3A+Running+Sneakers%0APrice%3A+%2459.99%0AInventory%3A+20%0A%0A HTTP/11" 200 459
2024-08-12 11:35:02,010 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222767&timeout=20 HTTP/11" 200 371
2024-08-12 11:35:02,156 - DEBUG - https://api.telegram.org:443 "POST /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/sendMessage?chat_id=245324256&text=Available+products%3A%0AID%3A+1%0AName%3A+Graphic+T-Shirt%0APrice%3A+%2419.99%0AInventory%3A+50%0A%0AID%3A+2%0AName%3A+Slim+Fit+Jeans%0APrice%3A+%2439.99%0AInventory%3A+30%0A%0AID%3A+3%0AName%3A+Running+Sneakers%0APrice%3A+%2459.99%0AInventory%3A+20%0A%0A HTTP/11" 200 459
2024-08-12 16:51:31,166 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 16:52:27,340 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 16:53:14,501 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 16:53:15,049 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=-1&timeout=20 HTTP/11" 200 371
2024-08-12 16:53:15,164 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getMe HTTP/11" 200 263
2024-08-12 16:53:15,164 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 16:53:15,697 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=1&timeout=20 HTTP/11" 200 720
2024-08-12 16:53:15,697 - ERROR - message_handler: Commands filter should be list of strings (commands), unknown type supplied to the 'commands' filter list. Not able to use the supplied type.
2024-08-12 16:53:15,702 - ERROR - message_handler: Commands filter should be list of strings (commands), unknown type supplied to the 'commands' filter list. Not able to use the supplied type.
2024-08-12 16:53:35,818 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222772&timeout=20 HTTP/11" 200 23
2024-08-12 16:54:08,580 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 16:54:10,628 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=-1&timeout=20 HTTP/11" 200 371
2024-08-12 16:54:10,737 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getMe HTTP/11" 200 263
2024-08-12 16:54:10,741 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 16:54:11,247 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=1&timeout=20 HTTP/11" 200 371
2024-08-12 16:54:11,247 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 16:54:11,842 - DEBUG - https://api.telegram.org:443 "POST /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/sendMessage?chat_id=245324256&text=Enter+the+category HTTP/11" 200 274
2024-08-12 16:54:20,917 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222773&timeout=20 HTTP/11" 200 304
2024-08-12 16:54:20,949 - ERROR - Infinity polling exception: module 'handlers.commands' has no attribute 'cid'
2024-08-12 16:54:20,949 - ERROR - Exception traceback:
Traceback (most recent call last):
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1090, in infinity_polling
    self.polling(non_stop=True, timeout=timeout, long_polling_timeout=long_polling_timeout,
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1178, in polling
    self.__threaded_polling(non_stop=non_stop, interval=interval, timeout=timeout, long_polling_timeout=long_polling_timeout,
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1253, in __threaded_polling
    raise e
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1215, in __threaded_polling
    self.worker_pool.raise_exceptions()
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\util.py", line 150, in raise_exceptions
    raise self.exception_info
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\util.py", line 93, in run
    task(*args, **kwargs)
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 8619, in _run_middlewares_and_handler
    if self._test_message_handler(handler, message):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 8548, in _test_message_handler
    if not self._test_filter(message_filter, filter_value, message):
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 8572, in _test_filter
    return filter_value(message)
           ^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\Personal Computer\Desktop\Ecommerce bot\ecommerce\bot.py", line 16, in <lambda>
    bot.message_handler(func=lambda m:commands.user_step.get(commands.cid)==0)(commands.get_product_category)
                                                             ^^^^^^^^^^^^
AttributeError: module 'handlers.commands' has no attribute 'cid'

2024-08-12 16:54:23,964 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 16:54:24,519 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222774&timeout=20 HTTP/11" 409 143
2024-08-12 16:54:44,522 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222774&timeout=20 HTTP/11" 200 23
2024-08-12 16:58:24,939 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 16:58:29,410 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=-1&timeout=20 HTTP/11" 200 371
2024-08-12 16:58:29,520 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getMe HTTP/11" 200 263
2024-08-12 16:58:29,520 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 16:58:30,080 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=1&timeout=20 HTTP/11" 200 371
2024-08-12 16:58:30,080 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 16:58:30,640 - DEBUG - https://api.telegram.org:443 "POST /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/sendMessage?chat_id=245324256&text=Enter+the+category HTTP/11" 200 274
2024-08-12 16:58:35,973 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222775&timeout=20 HTTP/11" 200 304
2024-08-12 16:58:36,226 - DEBUG - https://api.telegram.org:443 "POST /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/sendMessage?chat_id=245324256&text=Available+products%3A%0AID%3A+1%0AName%3A+Graphic+T-Shirt%0APrice%3A+%2419.99%0AInventory%3A+50%0A%0A HTTP/11" 200 335
2024-08-12 16:58:56,090 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222776&timeout=20 HTTP/11" 200 23
2024-08-12 16:59:06,281 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222776&timeout=20 HTTP/11" 200 371
2024-08-12 16:59:06,282 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 16:59:06,934 - DEBUG - https://api.telegram.org:443 "POST /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/sendMessage?chat_id=245324256&text=Enter+the+category HTTP/11" 200 274
2024-08-12 16:59:15,268 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222777&timeout=20 HTTP/11" 200 304
2024-08-12 16:59:15,428 - DEBUG - https://api.telegram.org:443 "POST /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/sendMessage?chat_id=245324256&text=Available+products%3A%0AID%3A+1%0AName%3A+Graphic+T-Shirt%0APrice%3A+%2419.99%0AInventory%3A+50%0A%0A HTTP/11" 200 335
2024-08-12 16:59:35,386 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222778&timeout=20 HTTP/11" 200 23
2024-08-12 16:59:55,521 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222778&timeout=20 HTTP/11" 200 23
2024-08-12 17:00:15,657 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222778&timeout=20 HTTP/11" 200 23
2024-08-12 17:00:35,789 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222778&timeout=20 HTTP/11" 200 23
2024-08-12 17:00:55,904 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222778&timeout=20 HTTP/11" 200 23
2024-08-12 17:01:16,020 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222778&timeout=20 HTTP/11" 200 23
2024-08-12 17:01:36,135 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222778&timeout=20 HTTP/11" 200 23
2024-08-12 17:01:56,249 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222778&timeout=20 HTTP/11" 200 23
2024-08-12 17:02:16,363 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222778&timeout=20 HTTP/11" 200 23
2024-08-12 17:02:36,484 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222778&timeout=20 HTTP/11" 200 23
2024-08-12 17:16:56,125 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:17:10,968 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=-1&timeout=20 HTTP/11" 200 361
2024-08-12 17:17:11,076 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getMe HTTP/11" 200 263
2024-08-12 17:17:11,078 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:17:11,597 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=1&timeout=20 HTTP/11" 200 361
2024-08-12 17:17:11,597 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:17:12,146 - DEBUG - https://api.telegram.org:443 "POST /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/sendMessage?chat_id=245324256&text=Available+commands%3A%0A%2Fstart+-+Welcome+message+and+basic+instructions%0A%2Fhelp+-+Display+this+help+message%0A%2Flist_products+-+List+products+by+category%0A%2Fproduct_detail+-+Product+detail+by+ID%0A%2Fview_cart+-+View+your+shopping+cart%0A%2Fadd_to_cart+%3Cproduct_id%3E+%3Cquantity%3E+-+Add+a+product+to+your+cart%0A%2Fremove_from_cart+%3Cproduct_id%3E+%3Cquantity%3E+-+Add+a+product+to+your+cart%0A%2Fprofile+-+View+or+update+your+profile+information%0A HTTP/11" 200 1064
2024-08-12 17:17:14,956 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222779&timeout=20 HTTP/11" 200 372
2024-08-12 17:17:15,107 - DEBUG - https://api.telegram.org:443 "POST /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/sendMessage?chat_id=245324256&text=Enter+the+product+ID HTTP/11" 200 276
2024-08-12 17:17:19,383 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222780&timeout=20 HTTP/11" 200 299
2024-08-12 17:17:19,415 - ERROR - Infinity polling exception: Could not process parameters: int(2), it must be of type list, tuple or dict
2024-08-12 17:17:19,415 - ERROR - Exception traceback:
Traceback (most recent call last):
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1090, in infinity_polling
    self.polling(non_stop=True, timeout=timeout, long_polling_timeout=long_polling_timeout,
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1178, in polling
    self.__threaded_polling(non_stop=non_stop, interval=interval, timeout=timeout, long_polling_timeout=long_polling_timeout,
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1253, in __threaded_polling
    raise e
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1215, in __threaded_polling
    self.worker_pool.raise_exceptions()
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\util.py", line 150, in raise_exceptions
    raise self.exception_info
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\util.py", line 93, in run
    task(*args, **kwargs)
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 8623, in _run_middlewares_and_handler
    result = handler['function'](message)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\Personal Computer\Desktop\Ecommerce bot\ecommerce\handlers\commands.py", line 64, in get_product_id
    product,img=get_product_detail(product_id)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\Personal Computer\Desktop\Ecommerce bot\ecommerce\utils\database_utils.py", line 32, in get_product_detail
    cursor.execute('select * from product where product_id=%s',product_id)
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\mysql\connector\cursor_cext.py", line 344, in execute
    prepared = self._connection.prepare_for_mysql(params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\mysql\connector\connection_cext.py", line 891, in prepare_for_mysql
    raise ProgrammingError(
mysql.connector.errors.ProgrammingError: Could not process parameters: int(2), it must be of type list, tuple or dict

2024-08-12 17:17:22,429 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:17:22,938 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222781&timeout=20 HTTP/11" 409 143
2024-08-12 17:18:02,602 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:18:04,782 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=-1&timeout=20 HTTP/11" 200 372
2024-08-12 17:18:04,888 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getMe HTTP/11" 200 263
2024-08-12 17:18:04,888 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:18:05,407 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=1&timeout=20 HTTP/11" 200 372
2024-08-12 17:18:05,410 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:18:05,991 - DEBUG - https://api.telegram.org:443 "POST /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/sendMessage?chat_id=245324256&text=Enter+the+product+ID HTTP/11" 200 276
2024-08-12 17:18:08,478 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222782&timeout=20 HTTP/11" 200 299
2024-08-12 17:18:08,481 - ERROR - Infinity polling exception: Could not process parameters: str(2), it must be of type list, tuple or dict
2024-08-12 17:18:08,481 - ERROR - Exception traceback:
Traceback (most recent call last):
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1090, in infinity_polling
    self.polling(non_stop=True, timeout=timeout, long_polling_timeout=long_polling_timeout,
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1178, in polling
    self.__threaded_polling(non_stop=non_stop, interval=interval, timeout=timeout, long_polling_timeout=long_polling_timeout,
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1253, in __threaded_polling
    raise e
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1215, in __threaded_polling
    self.worker_pool.raise_exceptions()
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\util.py", line 150, in raise_exceptions
    raise self.exception_info
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\util.py", line 93, in run
    task(*args, **kwargs)
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 8623, in _run_middlewares_and_handler
    result = handler['function'](message)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\Personal Computer\Desktop\Ecommerce bot\ecommerce\handlers\commands.py", line 64, in get_product_id
    product,img=get_product_detail(product_id)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\Personal Computer\Desktop\Ecommerce bot\ecommerce\utils\database_utils.py", line 32, in get_product_detail
    cursor.execute('select * from product where product_id=%s',product_id)
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\mysql\connector\cursor_cext.py", line 344, in execute
    prepared = self._connection.prepare_for_mysql(params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\mysql\connector\connection_cext.py", line 891, in prepare_for_mysql
    raise ProgrammingError(
mysql.connector.errors.ProgrammingError: Could not process parameters: str(2), it must be of type list, tuple or dict

2024-08-12 17:18:11,497 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:18:12,019 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222783&timeout=20 HTTP/11" 409 143
2024-08-12 17:18:32,023 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222783&timeout=20 HTTP/11" 200 23
2024-08-12 17:18:58,217 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:19:01,405 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=-1&timeout=20 HTTP/11" 200 372
2024-08-12 17:19:01,515 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getMe HTTP/11" 200 263
2024-08-12 17:19:01,515 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:19:02,060 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=1&timeout=20 HTTP/11" 200 372
2024-08-12 17:19:02,065 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:19:02,878 - DEBUG - https://api.telegram.org:443 "POST /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/sendMessage?chat_id=245324256&text=Enter+the+product+ID HTTP/11" 200 276
2024-08-12 17:19:05,642 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222784&timeout=20 HTTP/11" 200 299
2024-08-12 17:19:05,650 - ERROR - Infinity polling exception: Could not process parameters: str(1), it must be of type list, tuple or dict
2024-08-12 17:19:05,650 - ERROR - Exception traceback:
Traceback (most recent call last):
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1090, in infinity_polling
    self.polling(non_stop=True, timeout=timeout, long_polling_timeout=long_polling_timeout,
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1178, in polling
    self.__threaded_polling(non_stop=non_stop, interval=interval, timeout=timeout, long_polling_timeout=long_polling_timeout,
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1253, in __threaded_polling
    raise e
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 1215, in __threaded_polling
    self.worker_pool.raise_exceptions()
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\util.py", line 150, in raise_exceptions
    raise self.exception_info
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\util.py", line 93, in run
    task(*args, **kwargs)
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\telebot\__init__.py", line 8623, in _run_middlewares_and_handler
    result = handler['function'](message)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\Personal Computer\Desktop\Ecommerce bot\ecommerce\handlers\commands.py", line 64, in get_product_id
    product,img=get_product_detail(product_id)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\Personal Computer\Desktop\Ecommerce bot\ecommerce\utils\database_utils.py", line 32, in get_product_detail
    cursor.execute('select * from product where product_id=%s',product_id)
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\mysql\connector\cursor_cext.py", line 344, in execute
    prepared = self._connection.prepare_for_mysql(params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Personal Computer\Desktop\ClassExercise1\venv\Lib\site-packages\mysql\connector\connection_cext.py", line 891, in prepare_for_mysql
    raise ProgrammingError(
mysql.connector.errors.ProgrammingError: Could not process parameters: str(1), it must be of type list, tuple or dict

2024-08-12 17:19:08,667 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:19:09,209 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222785&timeout=20 HTTP/11" 409 143
2024-08-12 17:19:29,197 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222785&timeout=20 HTTP/11" 200 23
2024-08-12 17:19:49,380 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222785&timeout=20 HTTP/11" 200 23
2024-08-12 17:20:09,505 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222785&timeout=20 HTTP/11" 200 23
2024-08-12 17:20:29,623 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222785&timeout=20 HTTP/11" 200 23
2024-08-12 17:21:15,783 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:21:19,906 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=-1&timeout=20 HTTP/11" 200 372
2024-08-12 17:21:20,013 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getMe HTTP/11" 200 263
2024-08-12 17:21:20,013 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:21:20,565 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=1&timeout=20 HTTP/11" 200 372
2024-08-12 17:21:20,569 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:21:21,216 - DEBUG - https://api.telegram.org:443 "POST /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/sendMessage?chat_id=245324256&text=Enter+the+product+ID HTTP/11" 200 276
2024-08-12 17:21:23,466 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222786&timeout=20 HTTP/11" 200 299
2024-08-12 17:21:23,650 - DEBUG - https://api.telegram.org:443 "POST /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/sendMessage?chat_id=245324256&text=Product+detail%0A%282%2C+%27pants%27%2C+%27Slim+Fit+Jeans%27%2C+39.99%2C+30%2C+%27Stylish+slim+fit+jeans+with+a+modern+cut.%27%2C+datetime.datetime%282024%2C+8%2C+12%2C+11%2C+28%2C+40%29%29 HTTP/11" 200 402
2024-08-12 17:24:13,627 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:24:16,989 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=-1&timeout=20 HTTP/11" 200 372
2024-08-12 17:24:17,096 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getMe HTTP/11" 200 263
2024-08-12 17:24:17,099 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:24:17,665 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=1&timeout=20 HTTP/11" 200 372
2024-08-12 17:24:17,665 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:24:18,265 - DEBUG - https://api.telegram.org:443 "POST /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/sendMessage?chat_id=245324256&text=Enter+the+product+ID HTTP/11" 200 276
2024-08-12 17:24:21,406 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222788&timeout=20 HTTP/11" 200 299
2024-08-12 17:24:21,421 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:24:22,101 - DEBUG - https://api.telegram.org:443 "POST /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/sendMessage?chat_id=245324256&text=Product+detail%0Aproduct_id%3A+2category%3A+pantsname%3A+Slim+Fit+Jeansprice%3A+39.99inventory%3A+30description%3A+Stylish+slim+fit+jeans+with+a+modern+cut.added_date%3A+2024-08-12+11%3A28%3A40 HTTP/11" 200 430
2024-08-12 17:24:41,523 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222789&timeout=20 HTTP/11" 200 23
2024-08-12 17:25:32,108 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:25:33,818 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=-1&timeout=20 HTTP/11" 200 372
2024-08-12 17:25:33,945 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getMe HTTP/11" 200 263
2024-08-12 17:25:33,945 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:25:34,518 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=1&timeout=20 HTTP/11" 200 372
2024-08-12 17:25:34,521 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:25:35,097 - DEBUG - https://api.telegram.org:443 "POST /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/sendMessage?chat_id=245324256&text=Enter+the+product+ID HTTP/11" 200 276
2024-08-12 17:25:37,611 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222790&timeout=20 HTTP/11" 200 299
2024-08-12 17:25:37,791 - DEBUG - https://api.telegram.org:443 "POST /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/sendMessage?chat_id=245324256&text=Product+detail%0Aproduct_id%3A+2%0A%0Acategory%3A+pants%0A%0Aname%3A+Slim+Fit+Jeans%0A%0Aprice%3A+39.99%0A%0Ainventory%3A+30%0A%0Adescription%3A+Stylish+slim+fit+jeans+with+a+modern+cut.%0A%0Aadded_date%3A+2024-08-12+11%3A28%3A40%0A%0A HTTP/11" 200 454
2024-08-12 17:25:57,737 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222791&timeout=20 HTTP/11" 200 23
2024-08-12 17:27:10,895 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:27:12,368 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=-1&timeout=20 HTTP/11" 200 372
2024-08-12 17:27:12,478 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getMe HTTP/11" 200 263
2024-08-12 17:27:12,478 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:27:13,054 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=1&timeout=20 HTTP/11" 200 372
2024-08-12 17:27:13,054 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:27:13,631 - DEBUG - https://api.telegram.org:443 "POST /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/sendMessage?chat_id=245324256&text=Enter+the+product+ID HTTP/11" 200 276
2024-08-12 17:27:23,242 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222792&timeout=20 HTTP/11" 200 299
2024-08-12 17:27:23,410 - DEBUG - https://api.telegram.org:443 "POST /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/sendMessage?chat_id=245324256&text=Product+detail%0A%0Aproduct_id%3A+3%0A%0Acategory%3A+shoes%0A%0Aname%3A+Running+Sneakers%0A%0Aprice%3A+59.99%0A%0Ainventory%3A+20%0A%0Adescription%3A+Lightweight+and+durable+running+sneakers+for+athletes.%0A%0Aadded_date%3A+2024-08-12+11%3A28%3A51%0A%0A HTTP/11" 200 471
2024-08-12 17:28:34,701 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:28:39,980 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=-1&timeout=20 HTTP/11" 200 361
2024-08-12 17:28:40,097 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getMe HTTP/11" 200 263
2024-08-12 17:28:40,097 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:28:40,613 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=1&timeout=20 HTTP/11" 200 361
2024-08-12 17:28:40,613 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:28:41,149 - DEBUG - https://api.telegram.org:443 "POST /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/sendMessage?chat_id=245324256&text=Available+commands%3A%0A%2Fstart+-+Welcome+message+and+basic+instructions%0A%2Fhelp+-+Display+this+help+message%0A%2Flist_products+-+List+products+by+category%0A%2Fproduct_detail+-+Product+detail+by+ID%0A%2Fview_cart+-+View+your+shopping+cart%0A%2Fadd_to_cart+%3Cproduct_id%3E+%3Cquantity%3E+-+Add+a+product+to+your+cart%0A%2Fremove_from_cart+%3Cproduct_id%3E+%3Cquantity%3E+-+Add+a+product+to+your+cart%0A%2Fprofile+-+View+or+update+your+profile+information%0A%2Fadd_product+-+Add+product%0A%2Fremove_product+-+Remove+product%0A%2Fupdate_product+-+Update+product%0A HTTP/11" 200 1304
2024-08-12 17:50:39,896 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:50:47,730 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=-1&timeout=20 HTTP/11" 200 369
2024-08-12 17:50:47,839 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getMe HTTP/11" 200 263
2024-08-12 17:50:47,839 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:50:48,352 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=1&timeout=20 HTTP/11" 200 369
2024-08-12 17:50:48,359 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:50:48,897 - DEBUG - https://api.telegram.org:443 "POST /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/sendMessage?chat_id=245324256&text=To+add+product+use+this+format%3A+category%2Cname%2Cprice%2Cinventory%2Cdescription HTTP/11" 200 329
2024-08-12 17:51:08,476 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222795&timeout=20 HTTP/11" 200 23
2024-08-12 17:51:28,723 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222795&timeout=20 HTTP/11" 200 23
2024-08-12 17:51:48,879 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222795&timeout=20 HTTP/11" 200 23
2024-08-12 17:51:48,987 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222795&timeout=20 HTTP/11" 200 334
2024-08-12 17:51:49,005 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:51:49,727 - DEBUG - https://api.telegram.org:443 "POST /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/sendMessage?chat_id=245324256&text=Datas+have+been+added+successfully HTTP/11" 200 290
2024-08-12 17:52:01,874 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222796&timeout=20 HTTP/11" 200 371
2024-08-12 17:52:01,874 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:52:02,448 - DEBUG - https://api.telegram.org:443 "POST /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/sendMessage?chat_id=245324256&text=Enter+the+category HTTP/11" 200 274
2024-08-12 17:52:14,658 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222797&timeout=20 HTTP/11" 200 303
2024-08-12 17:52:14,670 - DEBUG - Starting new HTTPS connection (1): api.telegram.org:443
2024-08-12 17:52:15,204 - DEBUG - https://api.telegram.org:443 "POST /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/sendMessage?chat_id=245324256&text=Available+products%3A%0A%0AID%3A+3%0AName%3A+Running+Sneakers%0APrice%3A+%2459.99%0AInventory%3A+20%0A%0AID%3A+4%0AName%3A+addidas+shoe%0APrice%3A+%2455.0%0AInventory%3A+10%0A%0A HTTP/11" 200 396
2024-08-12 17:52:34,755 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222798&timeout=20 HTTP/11" 200 23
2024-08-12 17:52:54,881 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222798&timeout=20 HTTP/11" 200 23
2024-08-12 17:53:15,013 - DEBUG - https://api.telegram.org:443 "GET /bot7041414535:AAE46lwqg81SBPNxg6BsOO-jhQScP9BmUWQ/getUpdates?offset=224222798&timeout=20 HTTP/11" 200 23
